File: backend/src/SharedModels/Models/Lead.cs
namespace SharedModels;

public class Lead
{
    public string Name { get; set; } = string.Empty;
    public string Email { get; set; } = string.Empty;
    public string PhoneNumber { get; set; } = string.Empty;
    public string CompanyName { get; set; } = string.Empty;
}

public class LeadEvaluation
{
    public Lead Lead { get; set; } = new();
    public bool IsQualified { get; set; }
    public string? Reason { get; set; }
}

File: frontend/src/types/lead.ts
export interface LeadFormData {
  name: string;
  email: string;
  phoneNumber: string;
  companyName: string;
}

export interface Lead extends LeadFormData {
  createdAt?: string;
  status?: "new" | "contacted" | "qualified" | "disqualified";
}

export interface LeadEvaluation {
  lead: Lead;
  isQualified: boolean;
  reason: string;
}
